aoc2018/src/main/scala/day13.scala

```diff
@@ -1,8 +1,144 @@
-// IDEA: make list of Point that have + \ /
-// cards are tuple of (coords, dir, intersection) e.g. (Point(1,1), (1,0), (0,0,1)) --> moving right (x + 1, y + 0) each tick
-// / --> transforms (x+1, y-1)
-// \ --> transforms (x-1, y+1)
-// + --> transforms(x+0, y+0) or / \
-//
-// compute the whole lot. If any coord point of card is identical, then return crash coords
-// track cards only, + / \ are fixed and only operate if card has same coords
\ No newline at end of file
+import scala.io.*
+import math.*
+import scala.collection.mutable
+import scala.collection.mutable.Stack
+
+
+object day13 extends App:
+
+  private val day: String =
+    this.getClass.getName.drop(3).init
+
+  private val start1: Long =
+    System.currentTimeMillis
+
+  def rotateVector[A](n: Int, s: Vector[A]): Vector[A] =
+    if s.isEmpty then s
+    else
+      val nbound: Int = n % s.length // skipping the full rotation rounds
+      if nbound < 0 then rotateVector(nbound + s.length, s)
+      else s.drop(nbound) ++ s.take(nbound)
+
+  case class Point(x: Int, y: Int):
+    def add(p2: Point): Point = Point(x + p2.x, y + p2.y)
```

Comment:
note that symbolic definitions are also feasible - especially for coordinate use cases like this - eg. given two points `p1` and `p2` - `def +(p2: Point): Point = Point(x + p2.x, y + p2.y)` would yield the client side syntax `val p3 = p1 + p2`